### ISSUE TITLE
Clientside CSoundPatch:Stop() doesn't work if the entity the sound was attached to is invalid

### ISSUE THREAD

**Issue by thegrb93** [Facepunch/garrysmod-issues#3260]
If I do `test = CreateSound(Entity(143), 'mysound.wav') test:Play()`, and remove the entity, the sound will attach itself to another entity and continue to play and if I do `test:Stop()` the sound still continues to play.

**Comment by thegrb93** [comment]
It is a looping sound btw

**Comment by thegrb93** [comment]
I don't know if it's related or worth mentioning, but the 'fire burning' sound effect has this issue a lot when ignited props get removed.

**Comment by robotboy655** [comment]
Yeah we need to stop all sounds (of that entity) when an entity is removed and make the sound patches invalid, but its a little tricky to do considering the sounds are attached to entity ids.

**Comment by mcNuggets1** [comment]
Any news on this?
I've just encountered this problem and there is little to no workaround for this sadly.
I'd had to hook.Add EntityRemoved just to see for if the entity is being removed to cancel the sound.

**Comment by robotboy655** [comment]
Made it stop soundpatches if the entity is deleted, and also prevent soundpatches from playing for deleted entities.

This should solve this particular issue.

**Comment by Cpt-Hazama** [comment]
Rest in peace to all entities with CreateSound based death sounds 🥀

**Comment by robotboy655** [comment]
Why would you do that? Just use EmitSound?

**Comment by Cpt-Hazama** [comment]
No, you're right. This was just a blind-sided change that effected a lot of mods, especially SNPCs as they primarily use CSoundPatch for vocal sounds as we have more control/functions we can use on them. We were really confused when the wave of workshop users were reporting death sounds stopped playing entirely

edit: and just to clarify, this change is good, it was moreso the surprise factor from it
